sui-execution/suivm/sui-adapter/src/programmable_transactions/context.rs

```diff
@@ -1296,4 +1305,119 @@ mod checked {
             protocol_config,
         )
     }
+
+    // Implementation of the `DataStore` trait for the Move VM.
+    // When used during execution it may have a list of new packages that have
+    // just been published in the current context. Thos are used for module/type
+    // resolution when executing module init.
+    // It may be created with an empty slice of packages either when no publish/upgrade
+    // are performed or when a type is requested not during execution.
+    pub(crate) struct SuiDataStore<'state, 'a> {
+        linkage_view: &'a LinkageView<'state>,
+        new_packages: &'a Vec<MovePackage>,
+    }
+
+    impl<'state, 'a> SuiDataStore<'state, 'a> {
+        pub(crate) fn new(
+            linkage_view: &'a LinkageView<'state>,
+            new_packages: &'a Vec<MovePackage>,
+        ) -> Self {
+            Self {
+                linkage_view,
+                new_packages,
+            }
+        }
+
+        fn get_module(&self, module_id: &ModuleId) -> Option<&Vec<u8>> {
+            for package in self.new_packages {
```

Comment:
Didn't we use a map at some point to avoid this O(n)?

Reply:
we moved the max package publish/upgrade to be 5 and I did not want to touch the code considering it will be changing with the new loader